src/lib/storage/fsst_segment/fsst_segment_iterable.hpp

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <type_traits>
+
+#include "storage/segment_iterables.hpp"
+
+#include "storage/fsst_segment.hpp"
+#include "storage/vector_compression/resolve_compressed_vector_type.hpp"
+
+namespace opossum {
+
+template <typename T>
+class FSSTSegmentIterable : public PointAccessibleSegmentIterable<FSSTSegmentIterable<T>> {
+ public:
+  using ValueType = T;
+
+  explicit FSSTSegmentIterable(const FSSTSegment<T>& segment) : _segment{segment} {}
+
+  template <typename Functor>
+  void _on_with_iterators(const Functor& functor) const {
+    _segment.access_counter[SegmentAccessCounter::AccessType::Sequential] += _segment.size();
+
+    auto begin = Iterator{_segment, ChunkOffset{0}};
+    auto end = Iterator{_segment, static_cast<ChunkOffset>(_segment.size())};
+
+    functor(begin, end);
+  }
+
+  template <typename Functor, typename PosListType>
+  void _on_with_iterators(const std::shared_ptr<PosListType>& position_filter, const Functor& functor) const {
+    _segment.access_counter[SegmentAccessCounter::access_type(*position_filter)] += position_filter->size();
+
+    using PosListIteratorType = decltype(position_filter->cbegin());
+    auto begin =
+        PointAccessIterator<PosListIteratorType>{_segment, position_filter->cbegin(), position_filter->cbegin()};
+    auto end = PointAccessIterator<PosListIteratorType>{_segment, position_filter->cbegin(), position_filter->cend()};
+    functor(begin, end);
+  }
+
+  size_t _on_size() const { return _segment.size(); }
+
+ private:
+  const FSSTSegment<T>& _segment;
+
+ private:
+  class Iterator : public AbstractSegmentIterator<Iterator, SegmentPosition<T>> {
+   public:
+    explicit Iterator(const FSSTSegment<T>& segment, ChunkOffset chunk_offset)
+        : _segment{segment}, _chunk_offset{chunk_offset} {}
+
+   private:
+    friend class boost::iterator_core_access;  // grants the boost::iterator_facade access to the private interface
+
+    void increment() { ++_chunk_offset; }
+
+    void decrement() { --_chunk_offset; }
+
+    void advance(std::ptrdiff_t n) { _chunk_offset += n; }
+
+    bool equal(const Iterator& other) const { return _chunk_offset == other._chunk_offset; }
+
+    std::ptrdiff_t distance_to(const Iterator& other) const {
+      return std::ptrdiff_t{other._chunk_offset} - std::ptrdiff_t{_chunk_offset};
+    }
+
+    SegmentPosition<T> dereference() const {
+      std::optional<T> value = _segment.get_typed_value(_chunk_offset);
+      bool has_value = value.has_value();
+      return SegmentPosition<T>(has_value ? value.value() : T{}, !has_value, _chunk_offset);
+    }
+
+   private:
```

Comment:
Redundant private access specifier, as there is a `private:` before.

Reply:
We did it the way it was implemented in the other segments.

Dictionary:
https://github.com/hyrise/hyrise/blob/d50c5db23f1f53dfa5745706ea95cbc3b4f7a0a3/src/lib/storage/dictionary_segment/dictionary_segment_iterable.hpp#L110-L115

Frame Of Reference:
https://github.com/hyrise/hyrise/blob/d50c5db23f1f53dfa5745706ea95cbc3b4f7a0a3/src/lib/storage/frame_of_reference_segment/frame_of_reference_segment_iterable.hpp#L101-L106

LZ4:
https://github.com/hyrise/hyrise/blob/d50c5db23f1f53dfa5745706ea95cbc3b4f7a0a3/src/lib/storage/lz4_segment/lz4_segment_iterable.hpp#L135-L139

Reply:
See above.